numpy/core/tests/test_function_base.py

```diff
@@ -149,6 +150,14 @@ def test_dtype(self):
         assert_equal(y.dtype, dtype('int32'))
 
         # Native types
+        for x in range(1, 10):
+            y = geomspace(x, x**4, num=4, dtype=int)
+            assert_array_equal(y, [x, x**2, x**3, x**4])
```

Comment:
Does `num=9`, `assert_(y[::2] == [x, x**2, x**3, x**4])` work?

Reply:
I am confused. Did you mean
```
y = geomspace(x, x**9, num=9, dtype=int)
assert_(y[::2] == [x, x**3, x**5, x**7, x**9])
```

Reply:
Sorry, I can't count. I mean
```
geomspace(x, x**4, num=7, dtype=int)
assert_(y[::2] == [x, x**2, x**3, x**4])
```

Reply:
Sadly it doesn't! 😞 That is a very good test case. What happens to this PR now then?

Reply:
IMHO, while it is possible to detect `x` as the `base` in your test case, simply by checking:
```python
mod(log(stop) / log(start), 1) == 0  # is integer
``` 
However, it is next to impossible to detect `x` when `start=1` and `num` is arbitary. i.e. for function calls like `geomspace(1, 10 ** 6, num=100, dtype=int)` or `geomspace(1, 3 ** 6, num=27, dtype=int)` etc.